### DISCUSSION TITLE
Lesson 5: What is the purpose of this "New" In this contract and how come we are able to deploy contract from this ?

### DISCUSSION THREAD

**Discussion by ljb630** [Cyfrin/foundry-full-course-cu#1619] [General]
``// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

// This import can be used to take contract from other contract instedof 
// copying all the code and pasting in here as it makes big mess

import{firstcontract} from "./firststcontract.sol";
// This is used to deploye contract from other contract

contract StroageFactory{

    firstcontract public FirstContract;

    function simpleStorageFactorucontract() public {
        FirstContract = new firstcontract();

    }

}
`
`

This is the code but here's the thing i understood that we have given variable as **FirstContract** for our contract **firstcontract**
as its common to define but the thing that got confused me is this like **        FirstContract = new firstcontract();
** I don't know why but why are we using this **new following contract name** here?

**Comment by EngrPips** [comment]
```javascript
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

// This import can be used to take contract from other contract instedof
// copying all the code and pasting in here as it makes big mess

import{firstcontract} from "./firststcontract.sol";
// This is used to deploye contract from other contract

contract StroageFactory{

firstcontract public FirstContract;

function simpleStorageFactorucontract() public {
    FirstContract = new firstcontract();

}
}
```

**Reply by EngrPips** [reply]
Hello @ljb630, So whenever you see `firstContract = new FirstContract` know that right at that line of code, we are deploying a new contract, so basically what the line of code does is it create a new contract which is of type `FirstContract,` I hope I explained well enough to make you understand. If you still find anything confusing, please let me know so I can explain it better.

**Reply by ljb630** [reply]
So to create another contract we use this this  **new**?

**Reply by EngrPips** [reply]
Yeah, to create a contract, you use the `new` keyword
